modules/server_new/src/main/scala/observe/server/keywords/ConditionOps.scala

```diff
@@ -3,54 +3,153 @@
-    def backgroundLightStr: String = conditions.sb.map(_.label).getOrElse("Any")
-    def imageQualityDbl: Double = conditions.iq.map(_.toArcSeconds.value.toDouble).getOrElse(Double.NaN)
-    def cloudExtinctionDbl: Double = conditions.ce.map(_.toBrightness).getOrElse(Double.NaN)
-    def waterVaporDbl(site: Site): Double = conditions.wv.map(_.toMillimeters(site)).getOrElse(Double.NaN)
-    def backgroundLightDbl: Double = conditions.sb.map(_.toMicroVolts).getOrElse(Double.NaN)
-  }
-  
-  extension (cnd: ConstraintSet) {
-    def imageQualityStr: String = cnd.imageQuality.label
-    def cloudExtinctionStr: String = cnd.cloudExtinction.label
-    def waterVaporStr: String = cnd.waterVapor.label
-    def backgroundLightStr: String = cnd.skyBackground.label
-    def imageQualityDbl: Double = cnd.imageQuality.toArcSeconds.value.toDouble
-    def cloudExtinctionDbl: Double = cnd.cloudExtinction.toBrightness
-    def waterVaporDbl(site: Site): Double = cnd.waterVapor.toMillimeters(site)
-    def backgroundLightDbl: Double = cnd.skyBackground.toMicroVolts
+      case SkyBackground.Dark    => 20.7
+      case SkyBackground.Gray    => 19.5
+      case SkyBackground.Bright  => 18.0
+  }
+
+  private def nearestLookup[V](map: NonEmptyList[(Double, V)])(key: Double): V = {
+    @tailrec
+    def findAcc(acc: (Double, V), rem: List[(Double, V)]): V = rem match {
+      case Nil            => acc._2
+      case ::(head, next) =>
+        if (abs(acc._1 - key) <= abs(head._1 - key))
+          acc._2
+        else
+          findAcc(head, next)
+    }
+
+    findAcc(map.head, map.tail)
+  }
+
+  private def binnedLookup[A: Ordering, V](map: NonEmptyList[(A, V)])(key: A): V = {
+    @tailrec
+    def findAcc(acc: (A, V), rem: List[(A, V)]): V = rem match {
+      case Nil            => acc._2
+      case ::(head, next) =>
+        if (key <= head._1)
+          acc._2
+        else
+          findAcc(head, next)
+    }
+
+    findAcc(map.head, map.tail)
+  }
+
+  private val iqBins = NonEmptyList
+    .of(0.350, 0.475, 0.630, 0.780, 0.900, 1.02, 1.20, 1.65, 2.2, 3.4, 4.8, 11.7)
+    .zip(
+      NonEmptyList
+        .of(
+          NonEmptyList.of(0.60, 0.90, 1.20, 2.00),
+          NonEmptyList.of(0.60, 0.85, 1.10, 1.90),
+          NonEmptyList.of(0.50, 0.75, 1.05, 1.80),
+          NonEmptyList.of(0.50, 0.75, 1.05, 1.70),
+          NonEmptyList.of(0.50, 0.70, 0.95, 1.70),
+          NonEmptyList.of(0.40, 0.70, 0.95, 1.65),
+          NonEmptyList.of(0.40, 0.60, 0.85, 1.55),
+          NonEmptyList.of(0.40, 0.60, 0.85, 1.50),
+          NonEmptyList.of(0.35, 0.55, 0.80, 1.40),
+          NonEmptyList.of(0.35, 0.50, 0.75, 1.25),
+          NonEmptyList.of(0.35, 0.50, 0.70, 1.15),
+          NonEmptyList.of(0.31, 0.37, 0.45, 0.75)
+        )
+        .map(_.zip(NonEmptyList.of(20, 70, 85, 100)))
+    )
+
+  extension (iq: ImageQuality) {
+    def toPercentile(wv: Wavelength, elevation: Angle): Int = {
+      // Pickering approximation (thanks Andy Stephens)
```

Comment:
What is pickering?
It seems this should go to `lucuma-core`

Reply:
Pickering is a name. Ref: https://en.wikipedia.org/wiki/Air_mass_(astronomy)#Interpolative_formulas